File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/repo/SeaCreatureJson.java
package at.hannibal2.skyhanni.data.jsonobjects.repo;

import at.hannibal2.skyhanni.utils.LorenzRarity;
import com.google.gson.annotations.Expose;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Map;

public class SeaCreatureJson {

    public static Type TYPE = new TypeToken<Map<String, SeaCreatureJson.Variant>>() {
    }.getType();

    public static class Variant {
        @Expose
        public String chat_color;
        @Expose
        public Map<String, SeaCreature> sea_creatures;
    }

    public static class SeaCreature {
        @Expose
        public String chat_message;
        @Expose
        public int fishing_experience;
        @Expose
        public Boolean rare;
        @Expose
        public LorenzRarity rarity;
    }

}

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/repo/SeaCreatureJson.kt
package at.hannibal2.skyhanni.data.jsonobjects.repo

import at.hannibal2.skyhanni.utils.LorenzRarity
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.google.gson.reflect.TypeToken
import java.lang.reflect.Type

data class SeaCreatureJson(
    @Expose @SerializedName("chat_color") val chatColor: String,
    @Expose @SerializedName("sea_creatures") val seaCreatures: Map<String, SeaCreatureInfo>,
) {
    companion object {
        val TYPE: Type = object : TypeToken<Map<String?, SeaCreatureJson>>() {}.type
    }
}

data class SeaCreatureInfo(
    @Expose @SerializedName("chat_message") val chatMessage: String,
    @Expose @SerializedName("alternate_messages") val alternateMessages: List<String> = emptyList(),
    @Expose @SerializedName("fishing_experience") val fishingExperience: Int,
    @Expose val rare: Boolean = false,
    @Expose val rarity: LorenzRarity,
)
